### ISSUE TITLE
Business API doesn't work with business ids that contains non ascii characters.

### ISSUE THREAD

**Issue by razola** [Yelp/yelp-ruby#51]
**momofuku-má-pêche-new-york-2** is a valid business id I got from the Search API. When trying to use it to get more data via the Business API I get the following error:

`URI::InvalidURIError: URI must be ascii only "/v2/business/momofuku-m\u{e1}-p\u{ea}che-new-york-2"`

**Comment by mittonk** [comment]
(From similar bugs in other Yelp API clients: 
The `business_id` here doesn't into a `?term=pizza` style query string helpers, so usually needs more explicit encoding & escaping.
https://github.com/Yelp/yelp-api/issues/118 was one recent example, we'll probably need the Ruby equivalent here.)

**Comment by razola** [comment]
Aha got it. This did the trick: 

```
require "era"
ERB::Util.url_encode(business_id)
```

http://apidock.com/ruby/ERB/Util/url_encode

**Comment by mittonk** [comment]
Excellent, this seems like a good improvement for yelp-ruby.

**Comment by mittonk** [comment]
A different error message, same idea:
```
bad URI(is not URI?): /v2/business/momofuku-má-pêche-new-york-2 (URI::InvalidURIError)
```

**Comment by mittonk** [comment]
Minimal repro:
```
# -*- coding: utf-8 -*-                                                                                                        
# Yelp.client.configure ...
resp = Yelp.client.business('momofuku-má-pêche-new-york-2')
puts resp.business.name
```

**Comment by mittonk** [comment]
@razola : OK, we finally got this merged into release 2.1.4.  Thanks for the contribution!
